From f8371722e9a7cfa1d4a1df2e539b45b616d28b3a Mon Sep 17 00:00:00 2001
From: "Mauro (mdrjr) Ribeiro" <mauro.ribeiro@hardkernel.com>
Date: Wed, 31 Jan 2024 18:36:49 +0000
Subject: [PATCH] arm64: meson: odroid-go-ultra: add battery and charger nodes

[Lakka: updated for the new rk818_battery driver (ROCKNIX 0022):
 - constant-charge-current-max-microamp 2000000, the hardware minimum
   with the 10 mOhm sense resistor
 - add rockchip,sense-resistor-micro-ohms = <10000> to the charger node
 - drop the old vendor-style rockchip,rk818-battery subnode]
---
 .../amlogic/meson-g12b-odroid-go-ultra.dts    | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts
index be8219dbb9c3..7d547d2f0d28 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts
@@ -25,6 +25,26 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+	bat: battery {
+		compatible = "simple-battery";
+		voltage-max-design-microvolt = <4200000>;
+		voltage-min-design-microvolt = <3500000>;
+		charge-full-design-microamp-hours = <4000000>;
+		charge-term-current-microamp = <200000>;
+		constant-charge-current-max-microamp = <2000000>;
+		constant-charge-voltage-max-microvolt = <4200000>;
+		factory-internal-resistance-micro-ohms = <180000>;
+
+
+		ocv-capacity-celsius = <20>;
+		ocv-capacity-table-0 =	<4146950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
+					<3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
+					<3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
+					<3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
+					<3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
+					<3574170 0>;
+	};
+
 	codec_clk: codec-clk {
 		compatible = "fixed-clock";
 		clock-frequency = <12288000>;
@@ -410,6 +430,12 @@ rk818: pmic@1c {
 		vcc9-supply = <&vddao_3v3>;
 		boost-supply = <&vdd_sys>;
 
+		charger {
+			compatible = "rockchip,rk818-charger";
+			monitored-battery = <&bat>;
+			rockchip,sense-resistor-micro-ohms = <10000>;
+		};
+
 		regulators {
 			vddcpu_a: DCDC_REG1 {
 				regulator-name = "vddcpu_a";
-- 
2.47.3

